OutOfSchool/OutOfSchool.DataAccess/Repository/CodeficatorRepository.cs

```diff
@@ -36,6 +38,8 @@ public async Task<List<CodeficatorAddressDto>> GetFullAddressesByPartOfName(stri
     {
         int cityAmountIfNamePartIsEmpty = 100;
 
+        namePart = namePart.RemoveCharsByRegexPattern(Constants.ExceptUkrainianLettersDashSpaceRegex);
```

Comment:
(Performance) Please consider usage of compiled regexp (either static instance created with `RegexOptions.Compiled` or using code generation via `GeneratedRegex` attribute) OR get rid of regular expression (simple `StringBuilder` filled by `for` cycle).

Reply:
I’ve made benchmarks:
![image](https://github.com/ita-social-projects/OoS-Backend/assets/107883170/47c42212-5725-4345-9298-d1e5ce1a75f5)
The fastest method is `StringBuilder`, but it consumes much more memory than other.
We discussed with @DmyMi what approach we should use – fastest execution or low memory usage.
He suggested using the lowest memory usage approach because when we have a lot of users searching for their settlement, high memory usage could be a bottleneck.
Following this suggestion we should use `GeneratedRegex`. 
Should I create a separate static partial class for regular expressions that use `GeneratedRegex` attribute?
Thank you!

Reply:
Regarding **memory allocation and performance**. 
You could use `stackalloc`, that is always faster than string builder and for small strings doesn't allocate memory - see screenshot with code and benchmark. At same time I don't insist, compiled regexp could be used as well

![image](https://github.com/ita-social-projects/OoS-Backend/assets/85107137/2018c529-d7a0-414d-adf5-0630f335940f)

Regarding **place of that generated regexp**. "Exclude everything except space, Ukrainian chars and dash" looks to me as a pretty generic task, that could be reused, so could be part of `StringExtensions`. 
If you believe that it has to be part of `CodeficatorRepository` you could move it there. 
And I see no necessity to create separate class just for this method.